autopilot/core/terminal.py

```diff
@@ -279,89 +283,69 @@ def initUI(self):
-        # multiply by three to get the inner (file, etc.) bar, the top bar (min, maximize, etc)
-        # and then the very top system tray bar in ubuntu
-        #titleBarHeight = self.style().pixelMetric(QtWidgets.QStyle.PM_TitleBarHeight,
-        #                                          QtWidgets.QStyleOptionTitleBar(), self) * 3
-        title_bar_height = screensize.height()-winsize.height()
-
-        #titleBarHeight = bar_height*2
-        # finally our logo
-        logo_height = bar_height
-
-
-
-        winheight = winsize.height() - title_bar_height - logo_height  # also subtract logo height
-        winsize.setHeight(winheight)
-        self.max_height = winheight
-        self.setGeometry(winsize)
-        self.setSizePolicy(QtWidgets.QSizePolicy.Maximum, QtWidgets.QSizePolicy.Maximum)
-
-        # Set heights on control panel and data panel
-
-
-        # move to primary display and show maximized
-        primary_display = app.desktop().availableGeometry(0)
-        self.move(primary_display.left(), primary_display.top())
-        # self.resize(primary_display.width(), primary_display.height())
-        #
-        self.control_panel.setMaximumHeight(winheight)
-        self.data_panel.setMaximumHeight(winheight)
+        
+        ## Set window size
+        # The window size behavior depends on TERMINAL_WINSIZE_BEHAVIOR pref
+        # If 'remember': restore to the geometry from the last close
+        # If 'maximum': restore to fill the entire screen
+        # If 'moderate': restore to a reasonable size of (1000, 400) pixels
+        terminal_winsize_behavior = prefs.get('TERMINAL_WINSIZE_BEHAVIOR')
+        
+        # Set geometry according to pref
+        if terminal_winsize_behavior == 'maximum':
+            # Set geometry to available geometry
+            self.setGeometry(primary_display)
+            
+            # Set SizePolicy to maximum
+            self.setSizePolicy(
+                QtWidgets.QSizePolicy.Maximum, QtWidgets.QSizePolicy.Maximum)
+
+            # Move to top left corner of primary display
+            self.move(primary_display.left(), primary_display.top())
+
+            # Also set the maximum height of each panel
+            self.control_panel.setMaximumHeight(primary_display.height())
+            self.data_panel.setMaximumHeight(primary_display.height())
+
+        elif terminal_winsize_behavior == 'remember':
+            # Attempt to restore previous geometry
+            if self.settings.value("geometry") is None:
+                # It was never saved, for instance, this is the first time
+                # this app has been run
+                # So default to the moderate size
+                self.move(primary_display.left(), primary_display.top())
+                self.resize(1000, 400)                
+            else:
+                # It was saved, so restore the last geometry
+                self.restoreGeometry(self.settings.value("geometry"))
+        
+        else:
```

Comment:
minor style thing, make explicit and then fallback, will do. otherwise this looks good